src/python/dxpy/scripts/dx_build_app.py

```diff
@@ -211,6 +211,59 @@ def parse_destination(dest_str):
     return try_call(resolve_path, dest_str)
 
 
+def _check_suggestions(src_dir, publish=False):
+    """
+    Examines the specified dxapp.json file and warns about any
+    violations of suggestions guidelines.
+    """
+    dxapp_json_filename = os.path.join(src_dir, "dxapp.json")
+    app_spec = json.load(open(dxapp_json_filename))
+    if 'inputSpec' in app_spec:
+        for input_field in app_spec['inputSpec']:
+            if 'suggestions' in input_field:
+                for suggestion in input_field['suggestions']:
+                    if 'project' in suggestion:
+                        try:
+                            project = dxpy.api.project_describe(suggestion['project'])
+                            if project['restricted']:
+                                logger.warn('Warning, project {name} is restricted!'.format(name=project['name']))
+                        except dxpy.exceptions.DXAPIError as e:
+                            if e.code == 404:
+                                raise dxpy.app_builder.AppBuilderException(
+                                    'Suggested project {name} does not exist, or not accessible by user'.format(
+                                        name=suggestion['project']))
+                        if 'path' in suggestion:
```

Comment:
Is it possible that suggestions includes a `path`, but not a `project`? If that is the case, we would need to add verification for the path.